### PR TITLE
Adds more info to the repos command

### LINKED ISSUE #451: Extend !repo with other UQCS repos
Eg `!repo slack` and `!repo signup`

Perhaps also include a `!repo list` command that DMs the result the to the user

### PR DESCRIPTION
**Pull request by Guthers**
Closes #451. Adds more repos to the repos command, still defaults to uqcsbot. Provides -l to list all available repos

### CODE REVIEW

File: uqcsbot/scripts/repo.py [hunk 1]
```diff
@@ -1,10 +1,58 @@
 from uqcsbot import bot, Command
 
+repos = {
+    "coc": "https://github.com/UQComputingSociety/code-of-conduct",
+    "constitution": "https://github.com/UQComputingSociety/constitution",
+    "cookbook": "https://github.com/UQComputingSociety/cookbook",
+    "desgin": "https://github.com/UQComputingSociety/design",
```
Comment (nicklambourne) [inline uqcsbot/scripts/repo.py]: typo in design

File: uqcsbot/scripts/repo.py [hunk 2]
```diff
@@ -1,10 +1,58 @@
 from uqcsbot import bot, Command
 
+repos = {
+    "coc": "https://github.com/UQComputingSociety/code-of-conduct",
```
Comment (nicklambourne) [inline uqcsbot/scripts/repo.py]: Can this be explicitly "Code of Conduct"

File: uqcsbot/scripts/repo.py [hunk 3]
```diff
@@ -1,10 +1,58 @@
...
-    bot.post_message(command.channel_id, "UQ Computing Society :github: Repository\n"
-                     "https://github.com/UQComputingSociety/uqcsbot")
+    # Setup for message passing
+    channel = bot.channels.get(command.channel_id)
+    # Read the commands provided
+    command_args = command.arg.split() if command.has_arg() else []
+
+    # Checks for the list command
+    is_list_output = False
+    if '--list' in command_args:
```
Comment (nicklambourne) [inline uqcsbot/scripts/repo.py]: Could this be simplified into a single if `'-l' in command_args or '--l' in command args` ?

**Comment by Guthers** [comment]
Expected outputs
image

File: uqcsbot/scripts/repo.py [hunk 4]
```diff
@@ -1,10 +1,58 @@
...
+                                          " please use -l/--list to print" +
+                                          " the full list_")
+        else:
+            # Add each of the specified repos, don't prompt with -l
+            for c in command_args:
+                repo_strs.append(f"<{repos[c]}|{c}>")
+
+    # Send the message to the channel
+    bot.post_message(channel, "Click the link to go to to the repo: " +
+                              ", ".join(repo_strs))
```
Comment (nicklambourne) [inline uqcsbot/scripts/repo.py]: Feel like a new line would be more appropriate as a separator.

File: uqcsbot/scripts/repo.py [hunk 5]
```diff
@@ -1,10 +1,58 @@
...
+        # Add all repos formatted as links
+        for k, v in repos.items():
+            repo_strs.append(f"<{v}|{k}>")
+    else:
+        # Add only the uqcsbot as the default result
+        if len(command_args) == 0:
+            repo_strs.append(f"<{repos['uqcsbot']}|uqcsbot>")
+            if not is_list_output:
+                bot.post_message(channel, "_Note: the list is not complete," +
+                                          " please use -l/--list to print" +
```
Comment (nicklambourne) [inline uqcsbot/scripts/repo.py]: Code formatting for flags would be appropriate

File: uqcsbot/scripts/repo.py [hunk 6]
```diff
@@ -1,10 +1,58 @@
...
+        is_list_output = True
+    if '-l' in command_args:
+        command_args.remove('-l')
+        is_list_output = True
+
+    # Setup the empty list of formatted repo strings
+    repo_strs = []
+    if is_list_output:
+        # Add all repos formatted as links
+        for k, v in repos.items():
```
Comment (nicklambourne) [inline uqcsbot/scripts/repo.py]: Maybe `name, url` ?

File: uqcsbot/scripts/repo.py [hunk 7]
```diff
@@ -1,10 +1,58 @@
...
+        if len(command_args) == 0:
+            repo_strs.append(f"<{repos['uqcsbot']}|uqcsbot>")
+            if not is_list_output:
+                bot.post_message(channel, "_Note: the list is not complete," +
+                                          " please use -l/--list to print" +
+                                          " the full list_")
+        else:
+            # Add each of the specified repos, don't prompt with -l
+            for c in command_args:
+                repo_strs.append(f"<{repos[c]}|{c}>")
```
Comment (nicklambourne) [inline uqcsbot/scripts/repo.py]: What happens on a key error here?

File: uqcsbot/scripts/repo.py (hunk 2 above)
Comment (Guthers) [reply uqcsbot/scripts/repo.py]: As "code-of-conduct" or "code_of_conduct"?

File: uqcsbot/scripts/repo.py (hunk 4 above)
Comment (Guthers) [reply uqcsbot/scripts/repo.py]: I tried a new line initially and it looked weird (also newline with four spaces as indents)

File: uqcsbot/scripts/repo.py (hunk 7 above)
Comment (Guthers) [reply uqcsbot/scripts/repo.py]: I have added a message for this now

File: uqcsbot/scripts/repo.py (hunk 4 above)
Comment (Guthers) [reply uqcsbot/scripts/repo.py]: You can compare these
image

image

**Comment by Guthers** [comment]
W.r.t to adding descriptions I am not sure if it will provide any value to the command but will clutter it up more, when really people will go to the repo if they care

**Pull request merged**